bool set_loc
---
bool success